src/NNLSSolverInterface.jl

```diff
@@ -0,0 +1,102 @@
+module NNLSSolverInterface
+
+export QPSolver
+
+using NNLS: QPWorkspace, load!, solve!
+importall MathProgBase.SolverInterface
+
+immutable QPSolver <: AbstractMathProgSolver
+    workspace::QPWorkspace{Float64, Int}
+
+    QPSolver() = new(QPWorkspace{Float64, Int}(0, 0))
+end
+
+type QPModel <: AbstractLinearQuadraticModel
+    workspace::QPWorkspace{Float64, Int}
+    lb::Vector{Float64}
+    ub::Vector{Float64}
+    A::Matrix{Float64}
+    constr_lb::Vector{Float64}
+    constr_ub::Vector{Float64}
+    Q::Matrix{Float64}
+    q::Vector{Float64}
+    sense::Symbol
+    solution::Vector{Float64}
+    duals::Vector{Float64}
+
+    QPModel(work::QPWorkspace) = new(work)
+end
+
+LinearQuadraticModel(s::QPSolver) = QPModel(s.workspace)
+
+numvar(work::QPWorkspace) = size(work.G, 2)
+numconstr(work::QPWorkspace) = size(work.G, 1)
+
+numvar(m::QPModel) = numvar(m.workspace)
+numconstr(m::QPModel) = numconstr(m.workspace)
+
+function loadproblem!(m::QPModel, A, lb, ub, q, constr_lb, constr_ub, sense)
+    m.sense = sense
+    m.A = A
+    m.lb = lb
+    m.ub = ub
+    m.constr_lb = constr_lb
+    m.constr_ub = constr_ub
+    m.q = q
+end
+
+setquadobj!(m::QPModel, Q) = m.Q = Q
+setquadobj!(m::QPModel, rowidx, colidx, quadval) = m.Q = sparse(rowidx, colidx, quadval)
```

Comment:
This should:

1. Set both the upper triangular and the lower triangular element
2. Add to the existing `m.Q` instead of clobbering the whole thing.

See https://mathprogbasejl.readthedocs.io/en/latest/lpqcqp.html#setquadobj!, doc for second method.